x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -869,3 +896,42 @@ function pipeMigrations(...migrations: AlertMigration[]): AlertMigration {
   return (doc: SavedObjectUnsanitizedDoc<RawRule>) =>
     migrations.reduce((migratedDoc, nextMigration) => nextMigration(migratedDoc), doc);
 }
+
+/**
+ * This creates a migration map that applies search source migrations to legacy es query rules
+ */
+const getEsQueryAlertSearchSourceMigrations = (searchSourceMigrations: MigrateFunctionsObject) =>
+  mapValues<MigrateFunctionsObject, MigrateFunction>(
+    searchSourceMigrations,
+    (migrate: MigrateFunction<SerializedSearchSourceFields>): MigrateFunction =>
+      (state) => {
+        const _state = state as unknown as { attributes: RawRule };
+
+        const serializedSearchSource = _state.attributes.params
+          .searchConfiguration as SerializedSearchSourceFields;
+
+        if (!serializedSearchSource) return _state;
+
+        return {
+          ..._state,
+          attributes: {
+            ..._state.attributes,
+            params: {
+              ..._state.attributes.params,
+              searchConfiguration: migrate(serializedSearchSource),
+            },
+          },
+        };
+      }
+  );
+
+export const getAllMigrations = (
+  searchSourceMigrations: MigrateFunctionsObject,
+  alertingSavedObjectTypeMigrations: SavedObjectMigrationMap
+): SavedObjectMigrationMap =>
+  mergeSavedObjectMigrationMaps(
+    alertingSavedObjectTypeMigrations,
+    getEsQueryAlertSearchSourceMigrations(
+      searchSourceMigrations
+    ) as unknown as SavedObjectMigrationMap
+  );
```

Comment:
Please move these new migration into the versioned list in the original  `getMigration` function.

https://github.com/elastic/kibana/blob/173374dc2e6fee3b068438218c5f4f1279d5b710/x-pack/plugins/alerting/server/saved_objects/migrations.ts#L165-L180

We keep this versioned list as a single source of truth for every migration based on the version where they were introduced.
So by adding this new one in a different place we're making maintenance harder.

Reply:
@elastic/response-ops-ram  I think it's time we refactored this file so that each version has a dedicated source file.
We have too much code in one big, unwieldy, file.

I'll file an issue.

Reply:
New migration moved to `getMigration` function.